store/sqlstore/post_store.go

```diff
@@ -2364,7 +2365,54 @@ func (s *SqlPostStore) cleanupThreads(postId, rootId string, permanent bool) err
 		return nil
 	}
 	if rootId != "" {
-		_, err := s.GetMaster().Exec(`UPDATE Threads SET ReplyCount = ReplyCount - 1 WHERE PostId = :Id AND ReplyCount > 0`, map[string]interface{}{"Id": rootId})
+		queryString, args, _ := s.getQueryBuilder().
+			Select("COUNT(Id)").
+			From("Posts").
+			Where(sq.And{
+				sq.Eq{"RootId": rootId},
+				sq.Eq{"UserId": userId},
+				sq.Eq{"DeleteAt": 0},
+			}).
+			ToSql()
+
+		count, err := s.GetReplica().SelectInt(queryString, args...)
+
+		if err != nil {
+			mlog.Warn("Error counting user's posts in thread", mlog.Err(err))
+		}
```

Comment:
If this fails, then `count` will be `0` and the user will be removed from the participants array. Maybe we should skip removal if this fails. I'm 0/5.

Reply:
made the change :)